docs/developer/cloud_testing.md

```diff
@@ -0,0 +1,122 @@
+# Cloud testing
+
+## Usage
+
+### Overview
+
+A cloud deployment is required to test the integration with resources that cannot be run in a local emulator, such as Azure Data Factory.
+
+- A **GitHub environment** is used to encapsulate secrets.
+- An **application** is created to represent the system when running test. In Azure Active Directory, a service principal for the application is configured in the cloud tenant, and configured to trust the GitHub environment using Federated Identity Credentials. For running tests locally, developers should be assigned equivalent or higher permissions.
+- **Cloud resources** and **role assignments** are provisioned using Terraform. To simplify the configuration and reduce the risk of leakage of elevated credentials, Terraform is executed manually by privileged developers on their local machine (rather than in a GitHub workflow).
```

Comment:
Why we do not want to automate this by executing terraform on CI ? It will be good that it can deploy required infra service e.g. ADF and eventually destroy them using terraform or shell script (deleting entire resource group) at the end of test piepeline run.

Reply:
We could, but there's not much point in the extra time and complexity. It also incurs no cost.